### PR TITLE
🍰 First Implementation Of Filtering Posts By Language

### LINKED ISSUE #2045: 🚀 [Feature] Filter Posts by language
## :rocket: Feature

grafik

I think that entering HC as non German speaker may cause the impression of a pure German Network, because until now nearly all the posts you see are in German.
To avoid this, it should be possible to filter the post by language, and may be set the default of the filter to the selected language.

### Design & Layout
As indicated in the screenshot. May be it is better placed below the emotes.

### Additional context
Thinking about the possiibilty to save filters.

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest
grafik

First implementation.

### Issues

- fixes #2045

### Todo

- [ ] Finding a way to display flags of the languages. See comments of #2045 .
- [ ] Add tests.

### CODE REVIEW

**Comment by Tirokk** [comment]
Hey @Mogge !!! Great work! 🚀💫
Thanks a lot for your contribution. 👍🏼

That looks and functions very good in my opinion @Mogge . Even if some things are still missing.
Tests for example. I added a checkbox to the PRs description.

I was first not aware that I have to scroll down in the filters menu to see the buttons.

And it is kind of discussable if this is at least the right place to have it. Or should it be a general setting of the users profile? @datenbrei

File: webapp/locales/en.json [hunk 1]
```diff
@@ -87,7 +87,11 @@
     },
     "followers": {
       "label": "Users I follow"
-    }
+    },
+      "language": {
```
Comment (Tirokk) [inline webapp/locales/en.json]: Please format with correct indent. ☝🏼

File: webapp/locales/de.json [hunk 2]
```diff
@@ -86,7 +86,11 @@
     },
     "followers": {
       "label": "Benutzern, denen ich folge"
-    }
+    },
+      "language": {
```
Comment (Tirokk) [inline webapp/locales/de.json]: Please format with correct indent.

**Comment by Tirokk** [comment]
You have some build errors @Mogge .
Please click on details and then on `The Build`:
See https://travis-ci.com/Human-Connection/Human-Connection/builds/133810500

**Comment by Tirokk** [comment]
And the flags are not absolutely related to the flags of the countries as well. @Mogge @datenbrei 
English language should have the English or the USA flag as an example … ?

Additional we need related countries on the posts and be able to filter for them as well. See #782

**Comment by Mogge** [comment]
The problem with the flags is even greater in Spanish. I do not like flags at all. But it would be nice to have a graphical representation for each language, better than the flag icon I use now.
I will check the errors and try to write some tests.
I think it is more important to filter by language than by country by now. We want to have HC attractive for non-German speakers. When they enter the web and only see German post they may think it is only a German community. So may be by default we should show only posts in the chosen language to new users.

**Comment by datenbrei** [comment]
@Tirokk What would a country-specific post be? One about a country, one from a country? Related to a country? From a specific author of such a country? Or an author living in a country? 

First I would stick with just language, later first we have to decide, what we really want to do with it.

**Comment by datenbrei** [comment]
@Mogge and @Tirokk For now I'd take the english flag. It is the english language, spoken all over the world.

**Comment by Mogge** [comment]
I pushed a new version with the tests added. When I ran yarn test I had no errors.

File: webapp/components/FilterPosts/FilterPosts.spec.js [hunk 3]
```diff
@@ -97,13 +107,31 @@ describe('FilterPosts.vue', () => {
       expect(mutations['posts/TOGGLE_CATEGORY']).toHaveBeenCalledWith({}, 'cat4')
     })
 
+    it('calls TOGGLE_LANGUAGE when clicked', () => {
+      const wrapper = openFilterPosts()
+      englishButton = wrapper
+        .findAll('button.language-buttons')
+        .at(languages.findIndex(l => l.code === 'en'))
```
Comment (mattwr18) [inline webapp/components/FilterPosts/FilterPosts.spec.js]: I really like your way of doing this! one thing I complain about with vue-test-utils over enzyme for example is that there is not `filterWhere`... but I really like how you've done this... I need to refactor some of my tests that look for the button in a position, which is easy to break if someone adds another button, for example.

File: webapp/components/FilterPosts/LanguageFilterMenuItems.vue [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      <ds-flex-item
+        :width="{ base: '0%', sm: '0%', md: '0%', lg: '3%' }"
+        id="languages-menu-divider"
+      />
+      <ds-flex v-for="language in locales" :key="language.code" class="languages-menu">
+        <ds-flex class="languages-menu">
+          <ds-flex-item width="100%" class="language-menu-item">
+            <ds-button
+              class="language-buttons"
+              icon="flag"
```
Comment (mattwr18) [inline webapp/components/FilterPosts/LanguageFilterMenuItems.vue]: As I stated in the comments, I would prefer using the same country codes for these buttons as the `LocaleSwitch`, using flags of countries is problematic, as discussed, but also the same icon doesn't make a lot of sense to me either.

**Comment by Mogge** [comment]
grafik

As agreed with @mattwr18 I changed the flag icon to the language code.
A pinned post in a language that is not in the filter stays pinned and visible to the user.
I think about a solution for the language filter in the user profile page, as it is not supposed that a user changes the language filter a lot. The idea is adding a `languageFilter property to the user, which is always applied when filtering posts.

File: webapp/components/FilterPosts/LanguageFilterMenuItems.vue (hunk 4 above)
Comment (Mogge) [reply webapp/components/FilterPosts/LanguageFilterMenuItems.vue]: grafik

**Comment by roschaefer** [comment]
@mattwr18 are we good to go here?

**Comment by mattwr18** [comment]
I would be ok to merge this, but if I am being picky @Mogge, I do ask myself why the buttons are rectangular when all the other filter buttons are circular... maybe I am being pedantic... what do you think @alina-beck?

**Comment by alina-beck** [comment]
@mattwr18 @Mogge I thought the same when I saw this... We _are_ picky. :grin: But right now the style guide only allows `icons` for circular buttons so unless we want to save all language codes as svg icons I don't see that this can be solved. I am, however, planning to add this functionality when migrating the styleguide – and buttons will be one of the first things on my list! :)

**Pull request merged**